server/src/test/java/org/elasticsearch/cluster/allocation/ClusterRerouteIT.java

```diff
@@ -222,6 +222,11 @@ private void rerouteWithAllocateLocalGateway(Settings commonSettings) throws Exc
                 .setSettings(Settings.builder().put("index.number_of_shards", 1))
                 .execute().actionGet();
 
+        final boolean closed = randomBoolean();
+        if (closed) {
```

Comment:
This test executes a full cluster restart, so I adapted it to randomly run on closed indices in the same pull request.